### PR TITLE
survive nginx reload

### PR DESCRIPTION
**Pull request by lexelby**
We discovered that after 2 or 3 nginx reloads (SIGHUP), this module started crashing with SIGABRT.  The abort is in `ngx_http_auth_radius_module.c`, in `ngx_send_radius_request`.  `abort()` is being called because `radius_send_request()` returns NULL.

`radius_send_request()` returns NULL because `sendto()` returns errno 22, EINVAL, Invalid Argument.  `sendto()` is mad because the sockaddr passed has the value `0.0.0.0:0`, which is indeed an invalid argument.

The root cause for this is that `radius_servers` isn't cleared when nginx reloads.  New entries are added, but the old ones are kept around.  The old ones have references to the sockaddr produced by `ngx_parse_url`, but those addresses were allocated using `ngx_pcalloc`, which, as far as I understand it, allocates memory that is automatically freed (at some point; not sure when).  I suspect this memory is freed and then reallocated to someone else through another `ngx_pcalloc` call, which zeroes the memory.

Long story short: we need to empty the `radius_servers` list on config reload.  The easiest way I found to do that was to wipe the list at the time that the location configuration is generated.  I verified that before this fix, the number of entries in radius_servers would increase by the number of radius servers in the config every reload.  After, there is no longer an increase and we don't see the EINVAL/SIGABRT crashes.

This PR clears the radius_servers list in what I *think* is the right spot, and it also adds logging for the number of radius servers.

### CODE REVIEW

**Comment by xenols** [comment]
lexelby, thank you for research and fix this bug.

I think that it is better to clear the list of servers on the preconfiguration stage.
I made the appropriate commit:
https://github.com/xenols/nginx-radius/commit/4163f66d2e7f73fc623dc56ea62eac54beec6f2d
